engine/src/main/java/org/terasology/config/flexible/SettingImpl.java

```diff
@@ -36,7 +36,7 @@
  * @param <T> The type of the value this {@link SettingImpl} contains.
  */
 public class SettingImpl<T> implements Setting<T> {
-    private static final Logger LOGGER = LoggerFactory.getLogger(SettingImpl.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(Setting.class);
```

Comment:
A few of the changes in this PR are made to ensure that no one outside the `config.flexible` package knows about the existence of the `*Impl` classes (they will be made package-private and moved to a `config.flexible.internal` package in later changes in this PR).  The `SettingImpl` class is the de facto implementation for the `Setting` interface, but external users won't know that. Keeping that in mind, I thought it was better to log errors in `SettingImpl` as a `Setting`, since it is the default (and most likely _only_) implementation of `Setting`. However, looking at how other `Interface`-`InterfaceImpl` combos in the engine handle the logger class, I shall change it back to `SettingImpl` for consistency.

Reply:
I agree with the reasoning that it should log as Setting, but for consistency, SettingImpl works.
Legacy and consistency suck.